test/test_hpytype.py

```diff
@@ -288,3 +288,62 @@ def test_HPyDef_SET(self):
         assert p.y == 3
         p.z = 1075
         assert p.y == 5
+
+    def test_specparam_base(self):
+        mod = self.make_module("""
+            static HPyType_Spec Dummy_spec = {
+                .name = "mytest.Dummy",
+                .itemsize = 0,
+                .flags = HPy_TPFLAGS_DEFAULT | HPy_TPFLAGS_BASETYPE,
+            };
+
+            @INITCODE
+                HPyType_SpecParam param[] = {
+                    { HPyType_SpecParam_Base, ctx->h_LongType },
+                    { 0 }
+                };
+            @EXPORT_TYPE("Dummy", Dummy_spec, param)
+            @INIT
+        """)
+        assert isinstance(mod.Dummy, type)
+        assert mod.Dummy.__name__ == 'Dummy'
+        assert mod.Dummy.__module__ == 'mytest'
+        assert issubclass(mod.Dummy, int)
+        assert isinstance(mod.Dummy(), mod.Dummy)
+        assert mod.Dummy() == 0
+        assert mod.Dummy(42) == 42
+
+        class Sub(mod.Dummy):
+            pass
+        assert isinstance(Sub(), mod.Dummy)
+
+    def test_specparam_basestuple(self):
+        mod = self.make_module("""
+            static HPyType_Spec Dummy_spec = {
+                .name = "mytest.Dummy",
+                .itemsize = 0,
+                .flags = HPy_TPFLAGS_DEFAULT | HPy_TPFLAGS_BASETYPE,
+            };
+
+            @INITCODE
```

Comment:
personally, I find this `@INITCODE` usage very confusing: by reading the test it's not immediately clear what's happening, and seeing C code outside a function is ugly.
Some random suggestion for alternatives:

1. Don't use `@EXPORT_TYPE` at all; instead, we use `@EXTRA_INIT_FUNC` which takes a **function name** and calls it in the init; inside the function, we manually create the type and put it in the dict:
```C
void create_Dummy(HPyContext ctx, HPy mod) {
    HPyType_SpecParam params[] = { ... };
    HPy h_dummy = HPyType_FromSpec(..., params);
    HPy_SetAttr(ctx, mod, HPyUnicode_FromString(ctx, "Dummy", h_dummy);
}

@EXTRA_INIT_FUNC(create_Dummy)
@INIT // inside INIT, we cal all the EXTRA_INIT_FUNC registered above
```

2. use`@EXPORT_TYPE`, but in the 3rd argument pass a C function which *RETURNS* the `HPyType_SpecParam[]` array. This is probably more messy because you need to malloc/free the array, or to return a static which you fill at runtime, etc.

Reply:
"by reading the test it's not immediately clear what's happening": well, I'd argue that it's already not immediately clear what's happening, with all the other ``@`` custom-pseudo-macros.  But yes, your suggestion number 1 looks at least like progress.